File: src/main/java/Lec02/Lec02.java
package Lec02;

public class Lec02 {
    
    public static void main(String[] args) {
    
    }
    
    public boolean startsWithA1(String str) {
        if (str == null) {
            throw new IllegalArgumentException("null이 들어왔습니다.");
        }
        return str.startsWith("A");
    }
    
    public Boolean startsWithA2(String str) {
        if (str == null) {
            return null;
        }
        return str.startsWith("A");
    }
    
    public boolean startsWithA3(String str) {
        if (str == null) {
            return false;
        }
        return str.startsWith("A");
    }
}

File: src/main/kotlin/Lec02/Lec02.kt
package Lec02

fun main() {
    val str: String? = null
    println(str?.length ?: 0)

    println(startsWithA4("ABC"))
    println(startsWithA4(null))
}

fun startsWithA1(str: String?): Boolean {
    return str?.startsWith("A")
        ?: throw IllegalArgumentException("null이 들어왔습니다.")
}

fun startsWithA2(str: String?): Boolean? {
    return str?.startsWith("A")
}

fun startsWithA3(str: String?): Boolean {
    return str?.startsWith("A") ?: false
}

fun startsWithA4(str: String?): Boolean {
    return str!!.startsWith("A")
}
